main.nf

```diff
@@ -71,13 +76,53 @@ process synapse_get {
 
 }
 
+// Download files from SevenBridges
+process sbg_get {
 
-// Convert Synapse URIs and staged locations into sed expressions
-ch_synapse_files
-  .map { syn_uri, syn_id, syn_file -> /-e 's|\b${syn_uri}\b|${outdir}\/${syn_id}\/${syn_file.name}|g'/ }
-  .reduce { a, b -> "${a} ${b}" }
-  .set { ch_synapse_sed }
+  container "quay.io/biocontainers/sevenbridges-python:2.9.1--pyhdfd78af_0"
 
+  publishDir "${outdir}/${sbg_id}/", mode: 'copy'
+
+  secret 'SB_API_ENDPOINT'
+  secret 'SB_AUTH_TOKEN'
+
+  input:
+  tuple val(sbg_uri), val(sbg_id)   from ch_sbg_ids
+
+  output:
+  tuple val(sbg_uri), val(sbg_id), path("*")    into ch_sbg_files
+
+  when:
+  sbg_uris.size() > 0
+
+  script:
+  """
+
+  #!/usr/bin/env python3
+
+  import sevenbridges as sbg
+
+  api = sbg.Api()
+  download_file = api.files.get('${sbg_id}')
+  download_file.download(download_file.name)
+
+  """
+
+}
+// Mix channels, allowing for either of them to be null
+if (ch_synapse_files == null) {
+    ch_all_files = ch_sbg_files
+} else if (ch_sbg_files == null) {
+  ch_all_files = ch_synapse_files
+} else {
+  ch_all_files = ch_synapse_files.mix(ch_sbg_files)
+}
```

Comment:
I'm guessing that you got an error when you tried mixing one channel with another that was null? If so, that's a bummer. This approach is fine for this version.

Reply:
I had tried just mixing them originally and did receive an error. However, that must have been for a different reason at the time because when I tried just now it did work, so I am removing this if-else logic.